api/api/V1/Device.ts

```diff
@@ -699,4 +699,30 @@ export default function (app: Application, baseUrl: string) {
       });
     }
   );
+
+  /**
+     * @api {post} /api/v1/devices/heartbeat send device heartbeat
+     * @apiName heartbeat
+     * @apiGroup Device
+     *
+     * @apiUse V1DeviceAuthorizationHeader
+     *
+     * @apiBody {Date} nextHeartbeat time next heart beat is expected
+
+     * @apiUse V1ResponseSuccess
+     * @apiUse V1ResponseError
+     */
+  app.post(
+    `${apiUrl}/heartbeat`,
+    extractJwtAuthorisedDevice,
+    validateFields([body("nextHeartbeat").isISO8601().toDate()]),
+    async function (request: Request, response: Response, next: NextFunction) {
+      response.locals.requestDevice.updateHeartbeat();
```

Comment:
You don't appear to be using the user-supplied `request.body.nextHeartbeat` value when updating.